src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Private/JavaScriptMarshalImpl.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices.JavaScript.Private
+{
+    internal static partial class JavaScriptMarshalImpl
+    {
+        internal static Function log = new Function("data", @"console.log(data)");
+
+        static JavaScriptMarshalImpl()
+        {
+            toJsMarshallers.Add(typeof(IJSObject).TypeHandle.Value, MarshalJsObjectToJs);
+            toCsMarshallers.Add(typeof(IJSObject).TypeHandle.Value, MarshalJsObjectToCs);
+            toJsMarshallers.Add(typeof(string).TypeHandle.Value, MarshalStringToJs);
+            toCsMarshallers.Add(typeof(string).TypeHandle.Value, MarshalStringToCs);
+
+            //log.Call(null, $"JavaScriptMarshalImpl string {typeof(string).TypeHandle.Value}");
+            //log.Call(null, $"JavaScriptMarshalImpl void {typeof(void).TypeHandle.Value}");
+        }
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern InvokeJSResult InvokeJSFunction0(string internedFunctionName, IntPtr signature, IntPtr data);
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern InvokeJSResult InvokeJSFunction1(string internedFunctionName, IntPtr signature, IntPtr data);
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static unsafe extern string BindJSFunction(string internedFunctionName, int arg_count, IntPtr signature, out int function_js_handle, out int is_exception);
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern InvokeJSResult InvokeJSFunction(int function_js_handle, IntPtr data);
```

Comment:
It's not clear to me how this differs from InvokeJSFunctionN or why it has a 'data' pointer without a 'signature'.

Reply:
It probably needs better name. It's invoking function which was generated and bound already. I renamed it to `InvokeBoundJSFunction`